connector/saml/saml.go

```diff
@@ -334,19 +353,97 @@ func (p *provider) HandlePOST(s connector.Scopes, samlResponse string) (ident co
 	return ident, nil
 }
 
+func (p *provider) validateStatus(resp *response) error {
+	status := resp.Status
+	if status == nil {
+		return fmt.Errorf("response did not contain a Status")
+	}
+	statusCode := status.StatusCode
+	if statusCode == nil {
+		return fmt.Errorf("response did not contain a StatusCode")
+	}
+	if statusCode.Value != "urn:oasis:names:tc:SAML:2.0:status:Success" {
```

Comment:
Just looked at the [saml-schema-protocol-2.0.xsd](https://docs.oasis-open.org/security/saml/v2.0/saml-schema-protocol-2.0.xsd) and found that Status and the first level StatusCode are required on the Response.